enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentProcessingService.java

```diff
@@ -119,83 +119,142 @@ public List<DocumentVerificationEntity> submitDocuments(
+        final DocumentsSubmitResult results = submitDocumentToProvider(submittedDocuments, docVerificationsList, idVerification, ownerId);
+        processSubmitResults(results, docVerifications, ownerId);
 
-            // TODO - after synchronous submission to document verification provider the document state should be
-            // set to VERIFICATION_PENDING, for asynchronous processing the UPLOAD_IN_PROGRESS state should remain
+        docVerificationsList.stream()
+                .filter(doc -> StringUtils.isNotBlank(doc.getUploadId()))
+                .map(doc -> docMetadataMap.get(doc.getId()).getUploadId())
+                .filter(StringUtils::isNotBlank)
+                .forEach(fileUploadId -> {
+                    documentDataRepository.deleteById(fileUploadId);
+                    logger.info("Deleted stored document data with id={}, {}", fileUploadId, ownerId);
+                });
 
-            DocumentResultEntity docResult = createDocumentResult(docVerification, docSubmitResult);
-            docResult.setTimestampCreated(ownerId.getTimestamp());
+        return docVerificationsList;
+    }
 
-            docResults.add(docResult);
+    /**
+     * Process submit results.
+     * @param results Document submit result from provider.
+     * @param docVerificationsMap To pair results with corresponding DocumentVerificationEntity.
+     * @param ownerId Owner identification.
+     */
+    private void processSubmitResults(final DocumentsSubmitResult results,
+                                      final Map<String, DocumentVerificationEntity> docVerificationsMap,
+                                      final OwnerId ownerId) {
 
-            // Delete previously persisted document after a successful upload to the provider
-            if (docVerification.getUploadId() != null && docMetadata.getUploadId() != null) {
-                documentDataRepository.deleteById(docMetadata.getUploadId());
-            }
-        }
+        final List<DocumentResultEntity> docResults = new ArrayList<>();
 
-        documentVerificationRepository.saveAll(docVerifications);
+        for (final DocumentSubmitResult result : results.getResults()) {
+            final DocumentVerificationEntity docVerification = docVerificationsMap.get(result.getDocumentId());
+            processDocsSubmitResults(ownerId, docVerification, results, result);
 
-        for (int i = 0; i < docVerifications.size(); i++) {
-            DocumentVerificationEntity docVerificationEntity = docVerifications.get(i);
-            docResults.get(i).setDocumentVerification(docVerificationEntity);
+            final DocumentResultEntity docResult = createDocumentResult(docVerification, result);
+            docResult.setTimestampCreated(ownerId.getTimestamp());
+            docResults.add(docResult);
+            docResult.setDocumentVerification(docVerification);
```

Comment:
May we group the code logically and not mix it with each other? For example:
```suggestion
            final DocumentResultEntity docResult = createDocumentResult(docVerification, result);
            docResult.setTimestampCreated(ownerId.getTimestamp());
            docResult.setDocumentVerification(docVerification);
                        
            docResults.add(docResult);
```